src/compact_live.rs

```diff
@@ -370,19 +373,20 @@ pub trait FindexLiveCompact<
             )?;
 
             // Insert all recompacted chains.
-            self.insert_chain_table(&new_chains.iter().flat_map(|(_, v)| v.clone()).collect()).await?;
+            self.insert_chain_table(new_chains.iter().flat_map(|(_, v)| v.clone()).collect()).await?;
 
             // Try upserting the new Entry Table.
             let upsert_data = UpsertData::new(
                 &encrypted_entry_table,
                 noisy_entry_table.encrypt::<DEM_KEY_LENGTH, DemScheme>(k_value, rng)?,
             );
+            let n_upsert = upsert_data.len();
 
             // These are failures to upsert.
-            encrypted_entry_table = self.upsert_entry_table(&upsert_data).await?;
+            encrypted_entry_table = self.upsert_entry_table(upsert_data).await?;
 
             // Delete unused chains (at least one chain value per entry line):
-            let mut chains_to_delete = HashSet::with_capacity(upsert_data.len());
+            let mut chains_to_delete = Vec::with_capacity(n_upsert);
```

Comment:
You may create the `chains_to_delete` before calling `upsert_entry_table`. It avoids the `n_upsert` variable?

Reply:
done in ce7cf06